contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol

```diff
@@ -0,0 +1,611 @@
+    private s_consumers;
+
+  event SubscriptionCreated(uint64 indexed subscriptionId, address owner);
+  event SubscriptionFunded(uint64 indexed subscriptionId, uint256 oldBalance, uint256 newBalance);
+  event SubscriptionConsumerAdded(uint64 indexed subscriptionId, address consumer);
+  event SubscriptionConsumerRemoved(uint64 indexed subscriptionId, address consumer);
+  event SubscriptionCanceled(uint64 indexed subscriptionId, address to, uint256 amount);
+  event SubscriptionOwnerTransferRequested(uint64 indexed subscriptionId, address from, address to);
+  event SubscriptionOwnerTransferred(uint64 indexed subscriptionId, address from, address to);
+
+  error TooManyConsumers();
+  error InsufficientSubscriptionBalance();
+  error InvalidConsumer(uint64 subscriptionId, address consumer);
+  error ConsumerRequestsInFlight();
+  error InvalidSubscription();
+  error OnlyCallableFromLink();
+  error InvalidCalldata();
+  error MustBeSubOwner(address owner);
+  error PendingRequestExists();
+  error MustBeRequestedOwner(address proposedOwner);
+  error BalanceInvariantViolated(uint256 internalBalance, uint256 externalBalance); // Should never happen
+  event FundsRecovered(address to, uint256 amount);
+
+  mapping(bytes32 => uint96) /* id */ /* LINK balance */
+    private s_withdrawableTokens;
+
+  // ================================================================
+  // |                       Request state                          |
+  // ================================================================
+
+  struct Request {
+    address coordinator;
+    address client;
+    uint64 subscriptionId;
+    uint32 gasLimit;
+    uint96 estimatedCost;
+    uint256 timeoutTimestamp;
+    uint256 gasAfterPaymentCalculation;
+    uint96 adminFee;
+  }
+
+  mapping(bytes32 => Request) /* request ID */ /* Request data */
+    internal s_requests;
+
+  // ================================================================
+  // |                       Initialization                         |
+  // ================================================================
+  constructor(address link) {
+    LINK = LinkTokenInterface(link);
+  }
+
+  function getTotalBalance() external view returns (uint256) {
+    return s_totalBalance;
+  }
+
+  // ================================================================
+  // |                      Getter methods                          |
+  // ================================================================
+  function _isValidSubscription(uint64 subscriptionId) internal view {
+    if (s_subscriptions[subscriptionId].owner == address(0)) {
+      revert InvalidSubscription();
+    }
+  }
+
+  function _isValidConsumer(address client, uint64 subscriptionId) internal view {
+    if (s_consumers[client][subscriptionId].allowed == false) {
+      revert InvalidConsumer(subscriptionId, client);
+    }
+  }
+
+  function getCurrentsubscriptionId() external view returns (uint64) {
```

Comment:
minor capitalization typo: getCurrentSubscriptionId

Reply:
Removed as this was unused and we can derive it in other ways